src/transformers/models/patchtst/modeling_patchtst.py

```diff
@@ -0,0 +1,1958 @@
+                mask_value=config.mask_value,
+                seed_number=config.seed_number,
+            )
+        else:
+            self.masking = nn.Identity()
+        self.encoder = ChannelAttentionPatchTSTEncoder(config)
+
+        # Initialize weights and apply final processing
+        self.post_init()
+
+    def forward(
+        self,
+        past_values: torch.Tensor,
+        past_observed_mask: Optional[torch.Tensor] = None,
+        future_values: Optional[torch.Tensor] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+    ) -> Union[Tuple, PatchTSTModelOutputWithNoAttention]:
+        output_hidden_states = (
+            output_hidden_states if output_hidden_states is not None else self.config.output_hidden_states
+        )
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        if past_observed_mask is None:
+            past_observed_mask = torch.ones_like(past_values)
+
+        # x: tensor [bs x seq_len x in_channels]
+        scaled_past_values, loc, scale = self.scaler(past_values, past_observed_mask)
+
+        # patched_values: [bs x num_input_channels x num_patches x patch_length] for pretrain
+        patched_values = self.patching(scaled_past_values)
+        if self.mask_input:
+            masked_values, mask = self.masking(patched_values)
+        else:
+            masked_values, mask = self.masking(patched_values), None
+        encoder_output = self.encoder(masked_values, output_hidden_states=output_hidden_states)
+
+        hidden_states = encoder_output.last_hidden_state
+        encoder_states = encoder_output.hidden_states
+
+        if not return_dict:
+            return tuple(v for v in [hidden_states, encoder_states, patched_values, mask, loc, scale] if v is not None)
+        return PatchTSTModelOutputWithNoAttention(
+            last_hidden_state=hidden_states,
+            hidden_states=encoder_states,
+            patched_input=patched_values,
+            mask=mask,
+            loc=loc,
+            scale=scale,
+        )
+
+
+class MaskPretrainHead(nn.Module):
+    def __init__(self, config):
+        super().__init__()
+        self.dropout = nn.Dropout(config.dropout)
+        self.linear = nn.Linear(config.d_model, config.patch_length)
+        self.use_cls_token = config.use_cls_token
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        """
+        x: tensor [bs x nvars x num_patches x d_model]
+                or [bs x nvars x (num_patches+1) x d_model] if use cls_token
+        output: tensor [bs x nvars x num_patches x patch_length]
+        """
+        x = self.linear(self.dropout(x))  # [bs x nvars x num_patches x patch_length]
+        if self.use_cls_token:
+            x = x[:, :, 1:, :]  # remove the first cls token
+        return x
+
+
+class PatchTSTForMaskPretraining(PatchTSTPreTrainedModel):
```

Comment:
This class does not have a docstring yet like the other models

Reply:
Docstring is added.